### PR TITLE
chore: document use of web components with react

### LINKED ISSUE #23: Evaluate how Web Components work in common React environments
This is important for products built with Carbon v10 React, because the AI enhancements (e.g. slug, gradient) are only being built in Carbon v11 (React and Web Components).

If a product is built with Carbon v10 React, and they want an AI-enhanced modal component for example, we'd recommend they use the AI-enhanced Web Component modal (not React).

This POC is not applicable to products built with Carbon v11 React, as they would just use the AI-enhanced React components not the AI-enhanced Web Components.

### PR DESCRIPTION
**Pull request by lee-chase**
Closes #23 #25

Documents use of Carbon Web Components V2 (Carbon 11) in a Carbon 10 React application. 

#### Changelog

**New**

- docs/network-usage.png
- docs/prefix-changes.png

**Changed**

- docs/support-for-carbon-v10.md

#### Testing / Reviewing

Documentation, PR review should suffice.

Perhaps @jeffchew and @tay1orjones should take a look.

### CODE REVIEW

File: docs/support-for-carbon-v10.md [hunk 1]
```diff
@@ -218,14 +218,315 @@ ultimately results in additional tech debt to be paid off later.
 
 ## Use assets from IBM's Akamai CDN (hosted by IBM corporate webmaster team) in a v10 application
 
-This approach will be documented in more depth soon.
+Components published on a CDN can be in created/consumed in a number of forms.
+This section is based on the section
+[Using CDN](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/tree/main/packages/carbon-web-components#using-cdn)
+documented in the `Carbon Web Components` package. It builds a Raect application
```
Comment (jeffchew) [inline docs/support-for-carbon-v10.md:224]:
```suggestion
documented in the `Carbon Web Components` package. It builds a React application
```

File: docs/support-for-carbon-v10.md [hunk 2]
```diff
@@ -218,14 +218,315 @@ ultimately results in additional tech debt to be paid off later.
...
+but the technique is applicable to all reactive libraries (Angular, Vue, Svelte
+etc), as Web Components can be used in the same way as native HTML elements.
 
 ### Caveats
 
+The solution outlined here shows that use of Carbon 11 Web Components, via the
+CDN, is possible in a a Carbon v10 projects. In doing so it only makes use of a
```
Comment (jeffchew) [inline docs/support-for-carbon-v10.md:231]:
```suggestion
CDN, is possible in a Carbon v10 projects. In doing so it only makes use of a
```

File: docs/support-for-carbon-v10.md [hunk 3]
```diff
@@ -218,14 +218,315 @@ ultimately results in additional tech debt to be paid off later.
...
+export default function App() {
+.
+.
+.
+}
+```
+
+#### Verifying your prefix changes
+
+If you have multiple packages and dependencies you will need to ensure the all
```
Comment (jeffchew) [inline docs/support-for-carbon-v10.md:344]:
```suggestion
If you have multiple packages and dependencies you will need to ensure that all
```

File: docs/support-for-carbon-v10.md [hunk 4]
```diff
@@ -218,14 +218,360 @@ ultimately results in additional tech debt to be paid off later.
 
 ## Use assets from IBM's Akamai CDN (hosted by IBM corporate webmaster team) in a v10 application
 
-This approach will be documented in more depth soon.
+Components published on a CDN can be in created/consumed in a number of forms.
```
Comment (annawen1) [inline docs/support-for-carbon-v10.md:221]:
```suggestion
Components published on a CDN can be created/consumed in a number of forms.
```

File: docs/support-for-carbon-v10.md [hunk 5]
```diff
@@ -440,6 +439,14 @@ provide fill a slot as shown below.
...
 ```
 
+One key technology behind Web Components is the use of the Shadow DOM. The
+Shadow DOM is a hidden separate DOM tree attached to an element (referred to as
+the Shadow Host). This allows for the encapsulation of styles and functionality
+within the component. The Shadow DOM and its contained elements can be viewed
+with the browser inspector tool as shown.
+
+![Example of Shadow DOM in inspector tool ](shadow-dom-inspector.png)
+
```
Comment (jeffchew) [inline docs/support-for-carbon-v10.md:449]:
```suggestion
One key technology behind Web Components is the use of the [Shadow DOM](https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_shadow_DOM), which is a standard and part of the browser API available in all modern browsers today. The
Shadow DOM is a hidden separate DOM tree attached to an element (referred to as
the Shadow Host). This allows for the encapsulation of styles and functionality
within the component. The Shadow DOM and its contained elements can be viewed
with the browser inspector tool as shown.

![Example of Shadow DOM in inspector tool ](shadow-dom-inspector.png)

```

Sorry, added some content but might not be prettier friendly....

**Review by jeffchew — APPROVED** [review]

**Pull request merged**